goodreads/templates/goodreads/csv_upload.html

```diff
@@ -25,11 +47,11 @@
     {% endif %}
 <div class="pt-8" style="display: flex; justify-content: center;">
    <div class="border-2 border-black p-2 w-1/3">
-      <form action="" method="POST" enctype="multipart/form-data">
+      <form action="" method="POST" enctype="multipart/form-data" onsubmit= "upload()">
```

Comment:
if you use onsubmit instead of onclick, I think this will still submit the form and do a full page reload. Try this solution to prevent the default behavior.

https://stackoverflow.com/a/3350351